source/Projectile.cpp

```diff
@@ -219,14 +219,21 @@ void Projectile::Move(vector<Visual> &visuals, vector<Projectile> &projectiles)
 		if(weapon->InfraredTracking())
 			infraredConfused = Random::Real() > weapon->InfraredTracking();
 
-		// Optical: proportional tracking quality.
-		if(weapon->OpticalTracking())
-			opticalConfused = Random::Real() > weapon->OpticalTracking();
-
-		// Radar: If the target has no jamming, then proportional to tracking
+		// Optical and Radar: If the target has no jamming, then proportional to tracking
 		// quality. If the target does have jamming, then it's proportional to
 		// tracking quality, the strength of target's jamming, and the distance
 		// to the target (jamming power attenuates with distance).
+		if(weapon->OpticalTracking())
+		{
+			double opticalTracking = weapon->OpticalTracking();
+			double opticalJamming = target->Attributes().Get("optical jamming");
+			if(!opticalJamming)
+				opticalConfused = Random::Real() > opticalTracking;
+			else
+				opticalConfused = Random::Real() > (opticalTracking * position.Distance(target->Position()))
+					/ (sqrt(opticalJamming) * weapon->Range());
```

Comment:
Seems like this is the same as below, why not make move this into a separate function?

Reply:
done